### DISCUSSION TITLE
ILI9341 display with ESP32 8 bit parallel only displays stripes and half a screen

### DISCUSSION THREAD

**Discussion by PhilipGong** [Bodmer/TFT_eSPI#2744] [Q&A - General]
Hello, I am currently using the ILI9341 display controlled by an ESP32  DEVKITV1 board. However, the display only shows half a screen with stripes. I setup the display to run at 8 bit parallel mode, and below is the user_setup that I have.
```
TFT_eSPI ver = 2.5.0
Processor    = ESP32
Frequency    = 240MHz
Transactions = Yes
Interface    = Parallel
Display driver = 9341
Display width  = 240
Display height = 320

TFT_CS   = GPIO 27
TFT_DC   = GPIO 14
TFT_RST  = GPIO 26
TFT_WR   = GPIO 12
TFT_RD   = GPIO 13

TFT_D0   = GPIO 16
TFT_D1   = GPIO 4
TFT_D2   = GPIO 23
TFT_D3   = GPIO 22
TFT_D4   = GPIO 21
TFT_D5   = GPIO 19
TFT_D6   = GPIO 18
TFT_D7   = GPIO 17

Font GLCD   loaded
Font 2      loaded
Font 4      loaded
Font 6      loaded
Font 7      loaded
Font 8      loaded
Smooth font enabled
```
The images below shows my setup and the screen's display when running the TFT_graphicstest_one_lib
![IMG_5001](https://github.com/Bodmer/TFT_eSPI/assets/78095790/47ba6020-027b-45bc-a5dd-07187569f3f6)
![IMG_5003](https://github.com/Bodmer/TFT_eSPI/assets/78095790/c95eab30-102f-43ba-94b1-717485adfcc5)
![IMG_5002](https://github.com/Bodmer/TFT_eSPI/assets/78095790/f8b93ba1-e627-4ebc-a1e6-24185b098c49)

**Comment by Bodmer** [comment]
The display board is for a 16 bit interface. The library does not support that interface on the ESP32.

**Reply by PhilipGong** [reply]
Hello Bodmer, thanks for the reply. However on the display board there is an option to select 16 bit and 8 bit mode, as shown in the photo below. The seller that I got this from also states that it is 16/8 bit compatible, can I not use this with ESP32? Thanks.
![IMG_5004](https://github.com/Bodmer/TFT_eSPI/assets/78095790/90417249-d967-4326-861f-0a5a218859ec)

**Reply by Bodmer** [reply]
Those are not the mode settings I would expect. See here:
![20230715_111658](https://github.com/Bodmer/TFT_eSPI/assets/15824805/df08ef50-5842-4d05-bdf1-a3886334ed5f)

The 8 bit bus needs to be set to "Interface I" and data bits D0-7 used. Check the vendors page, maybe the display driver is not an ILI9341?
I have an 8 bit parallel ILI9341 working so when connected and mode pins are correct it should work.

**Reply by Bodmer** [reply]
As a test try using D0-7 on the display board and see what happens.

**Reply by PhilipGong** [reply]
Sorry for the late reply. I did try to use D0-7, but the screen only displayed a white screen without any "pattern" on it. Is there anyway for me to determine if I do get an ILI9341? By the way, I got this pinout as shown below, I checked it with a multimeter and it does seem to correspond to the pins on the FPC cable.
![image](https://github.com/Bodmer/TFT_eSPI/assets/78095790/4dcde29e-8b2e-41d9-b050-2229bd813df1)
